client/my-sites/upgrades/domain-management/domain-connect/domain-connect-authorize-footer.jsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import { actionType } from './constants';
+
+class DomainConnectAuthorizeFooter extends Component {
+	static propTypes = {
+		isPlaceholder: PropTypes.bool,
+		onClose: PropTypes.func,
+		onConfirm: PropTypes.func,
+		showAction: PropTypes.oneOf( actionType )
+	};
+
+	static defaultProps = {
+		isPlaceholder: false
+	};
+
+	placeholder = () => {
+		return (
+			<div className="domain-connect__is-placeholder">
+				<span></span>
+				<span></span>
+			</div>
+		);
+	}
+
+	renderActionConfirmCancel = () => {
+		const { translate } = this.props;
+		const notReadyToSubmit = actionType.READY_TO_SUBMIT !== this.props.showAction;
+
+		const confirm = translate( 'Confirm' );
+		const cancel = translate( 'Cancel' );
+
+		return (
+			<div>
+				<p>
+					{
+						translate( 'When you\'re ready to proceed, click %(confirm)s. If this isn\'t what you meant to do, ' +
+							'click %(cancel)s and we won\'t make any changes.', {
+								args: {
+									confirm: confirm,
+									cancel: cancel
+								}
+							} )
+					}
+				</p>
+				<Button
+					busy={ notReadyToSubmit }
+					className="domain-connect__button"
+					disabled={ notReadyToSubmit }
+					icon
+					onClick={ this.props.onConfirm }
+					primary>
+					<Gridicon icon="checkmark" /> { translate( 'Confirm' ) }
```

Comment:
Ugh. Yep. I meant to do that....that's actually why I pulled them out as `const`s.